### PR TITLE
using icon on card view headers

### PR DESCRIPTION
**Pull request by Kethen**
some previews
image
image
image
image
image
image

### CODE REVIEW

**Review by TheLastProject — CHANGES_REQUESTED** [review]
There are a few glitches, but mostly works great. I feel we should probably be able to display a bit of the picture below the header bar too though, but I spent the past... 4+ hours trying to get that to work and I just couldn't without breaking other parts of the UI so... low-prio anyway I guess.

This one though... I have no clue:

image

It happened with this image:
dog_hendrik_sticker_512

File: app/src/main/res/layout/loyalty_card_view_layout.xml [hunk 1]
```diff
@@ -244,50 +244,73 @@
...
+                    android:id="@id/toolbar"
+                    android:layout_width="fill_parent"
+                    android:layout_height="?actionBarSize"
+                    android:background="@android:color/transparent"
+                    android:theme="@style/CardView.ActionBarTheme"
+                    app:contentInsetStart="72.0dip"
+                    app:layout_collapseMode="pin" />
+
+            </com.google.android.material.appbar.CollapsingToolbarLayout>
+        </RelativeLayout>
```
[the remark this answers is not in the document]
Comment (Kethen) [reply app/src/main/res/layout/loyalty_card_view_layout.xml:313]: Setting the relative layout to invisible might solve it, I'll take a look later

**Comment by Kethen** [comment]
https://user-images.githubusercontent.com/1885159/141519249-5cbc3b06-60fe-4436-92f3-10733e945d74.png
I need more information on this one, the image sets fine on my emulator
Meanwhile I don't think poking the image out of the top bar looks good, but I can change it to that way if you think it looks better

**Comment by TheLastProject** [comment]
All I know is that the dog thing spams my Android Studio console:

```
D/onGlobalLayout: checking if bottom sheet size has to be changed
D/Catima: Button sheet should be 1448 pixels high
W/View: requestLayout() improperly called by androidx.appcompat.widget.AppCompatImageView{ed01abb V.ED..... ........ 0,0-1080,496 #7f0902c2 app:id/icon_image} during second layout pass: posting in next frame
D/onLayoutChange: collapsing toolbar change, height: 481
W/View: requestLayout() improperly called by androidx.appcompat.widget.AppCompatImageView{ed01abb V.ED..... ......ID 0,0-1080,496 #7f0902c2 app:id/icon_image} during layout: running second layout pass
D/onLayoutChange: collapsing toolbar change, height: 481
D/onGlobalLayout: checking if bottom sheet size has to be changed
D/Catima: Button sheet should be 1448 pixels high
W/View: requestLayout() improperly called by androidx.appcompat.widget.AppCompatImageView{ed01abb V.ED..... ........ 0,0-1080,496 #7f0902c2 app:id/icon_image} during second layout pass: posting in next frame
D/onLayoutChange: collapsing toolbar change, height: 481
W/View: requestLayout() improperly called by androidx.appcompat.widget.AppCompatImageView{ed01abb V.ED..... ......ID 0,0-1080,496 #7f0902c2 app:id/icon_image} during layout: running second layout pass
D/onLayoutChange: collapsing toolbar change, height: 481
D/onGlobalLayout: checking if bottom sheet size has to be changed
D/Catima: Button sheet should be 1448 pixels high
W/View: requestLayout() improperly called by androidx.appcompat.widget.AppCompatImageView{ed01abb V.ED..... ........ 0,0-1080,496 #7f0902c2 app:id/icon_image} during second layout pass: posting in next frame
D/onLayoutChange: collapsing toolbar change, height: 481
W/View: requestLayout() improperly called by androidx.appcompat.widget.AppCompatImageView{ed01abb V.ED..... ......ID 0,0-1080,496 #7f0902c2 app:id/icon_image} during layout: running second layout pass
D/onLayoutChange: collapsing toolbar change, height: 481
D/onGlobalLayout: checking if bottom sheet size has to be changed
D/Catima: Button sheet should be 1448 pixels high
W/View: requestLayout() improperly called by androidx.appcompat.widget.AppCompatImageView{ed01abb V.ED..... ........ 0,0-1080,496 #7f0902c2 app:id/icon_image} during second layout pass: posting in next frame
D/onLayoutChange: collapsing toolbar change, height: 481
W/View: requestLayout() improperly called by androidx.appcompat.widget.AppCompatImageView{ed01abb V.ED..... ......ID 0,0-1080,496 #7f0902c2 app:id/icon_image} during layout: running second layout pass
D/onLayoutChange: collapsing toolbar change, height: 481
D/onGlobalLayout: checking if bottom sheet size has to be changed
D/Catima: Button sheet should be 1448 pixels high
W/View: requestLayout() improperly called by androidx.appcompat.widget.AppCompatImageView{ed01abb V.ED..... ........ 0,0-1080,496 #7f0902c2 app:id/icon_image} during second layout pass: posting in next frame
D/onLayoutChange: collapsing toolbar change, height: 481
W/View: requestLayout() improperly called by androidx.appcompat.widget.AppCompatImageView{ed01abb V.ED..... ......ID 0,0-1080,496 #7f0902c2 app:id/icon_image} during layout: running second layout pass
D/onLayoutChange: collapsing toolbar change, height: 481
D/onGlobalLayout: checking if bottom sheet size has to be changed
D/Catima: Button sheet should be 1448 pixels high
W/View: requestLayout() improperly called by androidx.appcompat.widget.AppCompatImageView{ed01abb V.ED..... ........ 0,0-1080,496 #7f0902c2 app:id/icon_image} during second layout pass: posting in next frame
D/onLayoutChange: collapsing toolbar change, height: 481
W/View: requestLayout() improperly called by androidx.appcompat.widget.AppCompatImageView{ed01abb V.ED..... ......ID 0,0-1080,496 #7f0902c2 app:id/icon_image} during layout: running second layout pass
D/onLayoutChange: collapsing toolbar change, height: 481
D/onGlobalLayout: checking if bottom sheet size has to be changed
D/Catima: Button sheet should be 1448 pixels high
W/View: requestLayout() improperly called by androidx.appcompat.widget.AppCompatImageView{ed01abb V.ED..... ........ 0,0-1080,496 #7f0902c2 app:id/icon_image} during second layout pass: posting in next frame
D/onLayoutChange: collapsing toolbar change, height: 481
W/View: requestLayout() improperly called by androidx.appcompat.widget.AppCompatImageView{ed01abb V.ED..... ......ID 0,0-1080,496 #7f0902c2 app:id/icon_image} during layout: running second layout pass
D/onLayoutChange: collapsing toolbar change, height: 481
D/onGlobalLayout: checking if bottom sheet size has to be changed
D/Catima: Button sheet should be 1448 pixels high
W/View: requestLayout() improperly called by androidx.appcompat.widget.AppCompatImageView{ed01abb V.ED..... ........ 0,0-1080,496 #7f0902c2 app:id/icon_image} during second layout pass: posting in next frame
D/onLayoutChange: collapsing toolbar change, height: 481
W/View: requestLayout() improperly called by androidx.appcompat.widget.AppCompatImageView{ed01abb V.ED..... ......ID 0,0-1080,496 #7f0902c2 app:id/icon_image} during layout: running second layout pass
D/onLayoutChange: collapsing toolbar change, height: 481
D/onGlobalLayout: checking if bottom sheet size has to be changed
D/Catima: Button sheet should be 1448 pixels high
W/View: requestLayout() improperly called by androidx.appcompat.widget.AppCompatImageView{ed01abb V.ED..... ........ 0,0-1080,496 #7f0902c2 app:id/icon_image} during second layout pass: posting in next frame
D/onLayoutChange: collapsing toolbar change, height: 481
W/View: requestLayout() improperly called by androidx.appcompat.widget.AppCompatImageView{ed01abb V.ED..... ......ID 0,0-1080,496 #7f0902c2 app:id/icon_image} during layout: running second layout pass
D/onLayoutChange: collapsing toolbar change, height: 481
D/onGlobalLayout: checking if bottom sheet size has to be changed
D/Catima: Button sheet should be 1448 pixels high
W/View: requestLayout() improperly called by androidx.appcompat.widget.AppCompatImageView{ed01abb V.ED..... ........ 0,0-1080,496 #7f0902c2 app:id/icon_image} during second layout pass: posting in next frame
D/onLayoutChange: collapsing toolbar change, height: 481
W/View: requestLayout() improperly called by androidx.appcompat.widget.AppCompatImageView{ed01abb V.ED..... ......ID 0,0-1080,496 #7f0902c2 app:id/icon_image} during layout: running second layout pass
D/onLayoutChange: collapsing toolbar change, height: 481
D/onGlobalLayout: checking if bottom sheet size has to be changed
D/Catima: Button sheet should be 1448 pixels high
W/View: requestLayout() improperly called by androidx.appcompat.widget.AppCompatImageView{ed01abb V.ED..... ........ 0,0-1080,496 #7f0902c2 app:id/icon_image} during second layout pass: posting in next frame
D/onLayoutChange: collapsing toolbar change, height: 481
W/View: requestLayout() improperly called by androidx.appcompat.widget.AppCompatImageView{ed01abb V.ED..... ......ID 0,0-1080,496 #7f0902c2 app:id/icon_image} during layout: running second layout pass
D/onLayoutChange: collapsing toolbar change, height: 481
D/onGlobalLayout: checking if bottom sheet size has to be changed
D/Catima: Button sheet should be 1448 pixels high
W/View: requestLayout() improperly called by androidx.appcompat.widget.AppCompatImageView{ed01abb V.ED..... ........ 0,0-1080,496 #7f0902c2 app:id/icon_image} during second layout pass: posting in next frame
D/onLayoutChange: collapsing toolbar change, height: 481
W/View: requestLayout() improperly called by androidx.appcompat.widget.AppCompatImageView{ed01abb V.ED..... ......ID 0,0-1080,496 #7f0902c2 app:id/icon_image} during layout: running second layout pass
D/onLayoutChange: collapsing toolbar change, height: 481
D/onGlobalLayout: checking if bottom sheet size has to be changed
D/Catima: Button sheet should be 1448 pixels high
W/View: requestLayout() improperly called by androidx.appcompat.widget.AppCompatImageView{ed01abb V.ED..... ........ 0,0-1080,496 #7f0902c2 app:id/icon_image} during second layout pass: posting in next frame
D/onLayoutChange: collapsing toolbar change, height: 481
W/View: requestLayout() improperly called by androidx.appcompat.widget.AppCompatImageView{ed01abb V.ED..... ......ID 0,0-1080,496 #7f0902c2 app:id/icon_image} during layout: running second layout pass
D/onLayoutChange: collapsing toolbar change, height: 481
D/onGlobalLayout: checking if bottom sheet size has to be changed
D/Catima: Button sheet should be 1448 pixels high
W/View: requestLayout() improperly called by androidx.appcompat.widget.AppCompatImageView{ed01abb V.ED..... ......ID 0,0-1080,496 #7f0902c2 app:id/icon_image} during second layout pass: posting in next frame
```

> Meanwhile I don't think poking the image out below the top bar looks good, but I can change it to that way if you think it looks better

I think it does, but trust me, I spent like 4+ hours on this last evening, so I don't think it's worth the trouble :(

**Comment by Kethen** [comment]
image

**Review by TheLastProject — APPROVED** [review]

**Pull request merged**